sdk/src/main/java/io/opentelemetry/sdk/distributedcontext/DistributedContextSdk.java

```diff
@@ -151,7 +152,9 @@ public int hashCode() {
 
     @Override
     public DistributedContextSdk build() {
-      // TODO if (parent == null) parent = DistributedContextManager.getCurrentContext();
+      if (parent == null) {
```

Comment:
I've looked into this approach too - the problem with this is if a user called `setNoParent()`, that will set `parent` to null, but if at the same time there's a distributed context in the current context, `build()` will still return a child distributed context. I don't think this is what users wanted:
```java
try (Scope scope = contextManager.withContext(root)) {
  Builder builder = DistributedContextSdk.builder().setNoParent(); // I want another root.
  builder.build(); // this will give me a child. 
}
```

Reply:
I implemented it according to the spec as defined here:
https://github.com/open-telemetry/opentelemetry-java/blob/ed42237b6c71052bbffe4d849647c53fae81b246/api/src/main/java/io/opentelemetry/distributedcontext/DistributedContext.java#L76-L84

I already found the spec to be contradictory (setNoParent --> will use X as parent) and was wondering why `setNoParent` was there in first place then (if someone wouldn't want an explicit parent why would they call `setParent` at all just to call `setNoParent` afterwards?) but was assured it's correct and should stay there.

Looks like the second sentence should rather be removed from the spec, right?
Then I'd introduce a flag which is set using `setNoParent` and would prevent getting the implicit parent at `build()`.